taskchampion/lib/src/server.rs

```diff
@@ -180,6 +181,13 @@ pub unsafe extern "C" fn tc_server_new_gcp(
             //  - bucket ownership is transferred to this function
             let bucket = unsafe { TCString::val_from_arg(bucket) }.into_string()?;
 
+            // SAFETY:
+            //  - credentialpath is valid (promised by caller)
+            //  - credentialpath ownership is transferred to this function
+            let credentialpath = match credentialpath {
```

Comment:
Here I think you'll want to test whether `credentialpath == ""` and set the value passed to `ServerConfig::Gcp` to `None` for the empty string, and `Some(path)` otherwise.